somexchanger99/erp_utils.py

```diff
@@ -14,6 +14,58 @@ class ErpUtils(object):
     def __init__(self):
         self._client = Client(**settings.ERP_CONF)
 
+    def generate_e101_attachments(self, model, date, process, step, **kwargs):
+
+        date_to = kwargs.get('date_to')
+
+        polissa_obj = self._client.model('giscedata.polissa')
+        partner_obj = self._client.model('res.partner')
+        sw_obj = self._client.model('giscedata.switching')
+        sw_step_obj = self._client.model('giscedata.switching.step')
+        swe101 = self._client.model('giscedata.switching.e1.01')
+        switching_wizard = self._client.model('giscedata.switching.wizard')
+
+        e1s_objects = self._get_objects_with_attachment(
+            model, date, process=process, step=step, date_to=date_to
+        )
+        e101_attachments = []
+
+        for e1 in e1s_objects:
+            if not e1.polissa_ref_id:
+                continue
+
+            steps = sw_step_obj.search([('proces_id','=', 11)])
```

Comment:
only two things here
i would extract the number to a global constant like `E1_ID = 11`,
i would make this search out of the look because its constant and only necessary one time

Reply:
Fixed. retrieving the id from the model by name.